ivy/functional/frontends/jax/func_wrapper.py

```diff
@@ -88,3 +89,58 @@ def new_fn(*args, **kwargs):
 
 def to_ivy_arrays_and_back(fn: Callable) -> Callable:
     return outputs_to_frontend_arrays(inputs_to_ivy_arrays(fn))
+
+
+def to_ivy_dtype(dtype):
+    if not dtype or isinstance(dtype, str):
+        return dtype
+    if dtype in (int, float, bool) or ivy.is_native_dtype(dtype):
+        return ivy.as_ivy_dtype(dtype)
+    return np_frontend.to_ivy_dtype(dtype)._ivy_dtype
+
+
+def handle_jax_dtype(fn: Callable) -> Callable:
+    @functools.wraps(fn)
+    def new_fn(*args, dtype=None, **kwargs):
+        if len(args) > (dtype_pos + 1):
+            dtype = args[dtype_pos]
+            kwargs = {
+                **dict(
+                    zip(
+                        list(inspect.signature(fn).parameters.keys())[
+                            dtype_pos + 1 : len(args)
+                        ],
+                        args[dtype_pos + 1 :],
+                    )
+                ),
+                **kwargs,
+            }
+            args = args[:dtype_pos]
+        elif len(args) == (dtype_pos + 1):
+            dtype = args[dtype_pos]
+            args = args[:-1]
+
+        if not dtype:
+            return fn(*args, dtype=dtype, **kwargs)
+
+        dtype = to_ivy_dtype(dtype)
+        if not jax_frontend.config.jax_enable_x64:
+            dtype_replacement_dict = {
+                ivy.int64: ivy.int32,
+                ivy.uint64: ivy.uint32,
+                ivy.float64: ivy.float32,
+                "float64": "float32",
+                "uint64": "uint32",
+                "int64": "int32",
+            }
+            dtype = dtype_replacement_dict[dtype] \
+                if dtype in dtype_replacement_dict else dtype
+
+        if isinstance(dtype, str):
+            return fn(*args, dtype=dtype, **kwargs)
+        if isinstance(dtype, np_frontend.dtype):
+            return fn(*args, dtype=dtype._ivy_dtype, **kwargs)
+        return fn(*args, dtype=np_frontend.to_ivy_dtype(dtype), **kwargs)
```

Comment:
don't need these lines, conversion is already handled earlier, just `return fn(*args, dtype=dtype, **kwargs)`